modules/snooker/src/main/ts/ephox/snooker/api/Generators.ts

```diff
@@ -1,21 +1,25 @@
-import { Arr, Fun, Optional, Optionals, Singleton } from '@ephox/katamari';
+import { Arr, Fun, Optional, Optionals } from '@ephox/katamari';
 import { Attribute, Css, SugarElement, SugarNode } from '@ephox/sugar';
 
 import { getAttrValue } from '../util/CellUtils';
 
-export interface CellSpan {
+export interface RowData {
+  readonly element: SugarElement<HTMLTableRowElement | HTMLTableColElement>;
+}
+
+export interface CellData {
   readonly element: SugarElement<HTMLTableCellElement | HTMLTableColElement>;
   readonly colspan: number;
   readonly rowspan: number;
 }
 
 export interface Generators {
-  readonly cell: (cellSpan: CellSpan) => SugarElement<HTMLTableCellElement>;
-  readonly row: () => SugarElement<HTMLTableRowElement>;
+  readonly cell: (cellData: CellData) => SugarElement<HTMLTableCellElement>;
+  readonly row: (rowData: RowData) => SugarElement<HTMLTableRowElement>;
```

Comment:
Worth noting this row data is actually not used currently, as when cloning a row we've always just ignored any styles, classes, etc... That is likely something we'll want to revisit at some stage  (e.g. there's a regression in the GitHub backlog for classes being lost when inserting new rows) and it was a good time to include this while refactoring.